packages/dm-core-plugins/src/role_filter/RoleFilterPlugin.tsx

```diff
@@ -0,0 +1,96 @@
+import { IUIPlugin, ViewCreator } from '@development-framework/dm-core'
+import React, { useEffect, useState } from 'react'
+import { intersection } from 'lodash'
+import { Autocomplete, Banner, Icon } from '@equinor/eds-core-react'
+
+export const RoleFilterPlugin = (props: IUIPlugin): JSX.Element => {
+  const { idReference, config } = props
+  const [allowedViewConfigs, setAllowedViewConfigs] = useState<filteredView[]>(
+    []
+  )
+  const [openViewConfigs, setOpenViewConfigs] = useState<filteredView[]>([])
+  const [selectedRole, setSelectedRole] = useState<string>()
+  const [allowedRoles, setAllowedRoles] = useState<string[]>([])
+
+  // TODO: Implement real roles when User login is in place
+  const User = {
+    roles: ['admin', 'operator'],
+  }
+  type filteredView = {
```

Comment:
The type should be `PascalCase`. Also, maybe move it outside RoleFilterPlugin, just before?